gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -3597,13 +3599,123 @@ MacroExpander::match_token (Parser<MacroInvocLexer> &parser, AST::Token &token)
   return parser.skip_token (token.get_id ());
 }
 
+bool
+MacroExpander::match_n_matches (
+  Parser<MacroInvocLexer> &parser,
+  std::vector<std::unique_ptr<AST::MacroMatch>> &matches, ssize_t &match_amount,
+  ssize_t lo_bound, ssize_t hi_bound)
```

Comment:
do we need ssize_t? can we not use size_t?

Reply:
I was trying to indicate that `0` would be valid but we could still have a `-1` value as sentinel and to indicate that the parameter was unspecifed. But in the end it doesn't really matter as no repetition rule would do something like "between 0 and 0" so I think zero is fine as a sentinel. I've changed it to `size_t`

Reply:
I see what you mean I think if we can get away with representing it as zero and use size_t I think it makes it simpler. I am happy to be proved wrong down the line for a case I didn't consider fully.